File: 69.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 0, x
        while(l <= r):
            m = (l + r)//2
            val = m * m
            if val == x:
                return m
            elif val > x:
                r = m - 1
            else:
                l = m + 1
        return l - 1

File: 69.cpp
class Solution {
public:
    int mySqrt(int x) {
        int l = 0, r = x, m;
        unsigned long long val;

        while(l <= r) {
            m = l + (r - l)/2;
            val = (unsigned long long) m * m;
            if(val == x){
                return m;
            } else if (val > x) {
                r = m - 1;
            } else {
                l = m + 1;
            }
        }
        return l - 1;
    }
};